### PR TITLE
Flyway integration

### PR DESCRIPTION
**Pull request by ameenhere**

### CODE REVIEW

**Comment by ameenhere** [comment]
Continuous integration now fails due to Unit tests using H2 database. Flyway initial schema scripts are getting executed and fails.

File: dhis-2/dhis-support/dhis-support-db-migrations/src/main/java/org/hisp/dhis/db/migration/base/V2_30_0__Populate_Schema_On_Empty_Db.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                    boolean nonEmptyDatabase = rows.getBoolean( 1 );
+                    if ( !nonEmptyDatabase )
+                    {
+                        Connection mConnection = context.getConnection();
+                        ScriptRunner runner = new ScriptRunner( mConnection, false, true );
+                        Resource resource = new ClassPathResource( BASE_SCHEMA_SQL_LOCATION );
+                        InputStream resourceInputStream = resource.getInputStream();
+                        runner.runScript( new BufferedReader( new InputStreamReader( resourceInputStream ) ) );
+
+                        String generateUidFunctionSql = "create or replace function generate_uid()\n" + "  returns text as\n" + "$$\n" + "declare\n"
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-db-migrations/src/main/java/org/hisp/dhis/db/migration/base/V2_30_0__Populate_Schema_On_Empty_Db.java]: Can this be made more readable or externalized to a file?

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/base/V2_30_0__Populate_Schema_On_Empty_Db.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ */
+public class V2_30_0__Populate_Schema_On_Empty_Db extends BaseJavaMigration
+{
+
+    private final static String BASE_SCHEMA_SQL_LOCATION = "/org/hisp/dhis/db/base/dhis2_base_schema.sql";
+
+    public void migrate( Context context )
+        throws Exception
+    {
+        try (Statement select = context.getConnection().createStatement())
```
Comment (stian-sandvold) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/base/V2_30_0__Populate_Schema_On_Empty_Db.java]:
```suggestion
        try ( Statement select = context.getConnection().createStatement() )
```

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/base/V2_30_0__Populate_Schema_On_Empty_Db.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            try (ResultSet rows = select.executeQuery( "SELECT EXISTS( SELECT * FROM information_schema.tables  WHERE table_name = 'organisationunit');" ))
```
Comment (stian-sandvold) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/base/V2_30_0__Populate_Schema_On_Empty_Db.java]:
```suggestion
            try ( ResultSet rows = select.executeQuery( "SELECT EXISTS( SELECT * FROM information_schema.tables  WHERE table_name = 'organisationunit');" ) )
```

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v31/V2_31_4__Sample_java_migration.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.flywaydb.core.api.migration.Context;
+import java.sql.Statement;
+
+public class V2_31_4__Sample_java_migration extends BaseJavaMigration
+{
+    
+    public void migrate( Context context )
+        throws Exception
+    {
+        try (Statement select = context.getConnection().createStatement())
```
Comment (stian-sandvold) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v31/V2_31_4__Sample_java_migration.java]:
```suggestion
        try ( Statement select = context.getConnection().createStatement() )
```

**Pull request merged**